presentation/src/main/java/org/cryptomator/presentation/presenter/VaultListPresenter.kt

```
@@ -526,16 +534,140 @@ class VaultListPresenter @Inject constructor( //
+	}
+
+	private fun applicationComponent(): ApplicationComponent {
+		return (activity().application as CryptomatorApp).component
+	}
+
+	private fun parseJsonArray(json: String?): List<String> {
+		if (json.isNullOrEmpty() || json == "[]") return emptyList()
+		return try {
+			val jsonArray = JSONArray(json)
+			(0 until jsonArray.length()).map { jsonArray.getString(it) }
+		} catch (e: JSONException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)